### DISCUSSION TITLE
Document.tobytes() maybe not convert it to binary data???

### DISCUSSION THREAD

**Discussion by jscodecode** [pymupdf/PyMuPDF#2236] [Looking for help]
This is my code:
doc = fitz.open('test.pdf')
doc = doc.tobytes()
doc = doc.decode(encoding='utf-8')
Threr is an error:
'utf-8' codec can't decode byte 0x9c in position 521: invalid start byte.

I see the interface.
tobytes(garbage=0, clean=False, deflate=False, deflate_images=False, deflate_fonts=False, ascii=False, expand=0, linear=False, pretty=False, no_new_id=False, encryption=PDF_ENCRYPT_NONE, permissions=-1, owner_pw=None, user_pw=None)
ascii (bool) – convert binary data to ASCII.

Document.tobytes() maybe not convert it to binary data???
How can I solve this problems?

**Comment by JorjMcKie** [comment]
Here is a major misconception: `Document.tobytes()` is a PDF in memory, not text. Therefore it **_must_** be binary.
What do you want to achieve? Text maybe?

**Reply by jscodecode** [reply]
yes. I need to change it to text in utf-8，but my code is error in “doc.decode（encoding=“utf-8”）.
I think the reason is bytes of document.
There are some bytes，which are  not binary like “0x9C” in error tips.
How can i solve this problem？Thank you very much！

**Comment by JorjMcKie** [comment]
Well you must extract the text. This will be in UTF-8 encoding, so should be a no-brainer:
```python
doc = fitz.open("tst.pdf")
text = chr(12).join([page.get_text(sort=True) for page in doc])
```
The result, `text` is a UTF-8 string with formfeed character (chr(12)) between pages.

**Reply by JorjMcKie** [reply]
The `tobytes()` method is something entirely different: it does the same thing as `doc.save()` except it won't write to disk but to memory.

**Reply by jscodecode** [reply]
yes. I  konw that by reading  your reffernce of pymupdf  in website.
